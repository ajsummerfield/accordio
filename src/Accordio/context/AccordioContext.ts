import React from 'react';

interface AccordioContextValue {
  activeIndex: number;
  setActiveIndex: (index: number) => void;
}

const contextValue: AccordioContextValue = {
  activeIndex: -1,
  setActiveIndex: (index: number) => {}
};

const AccordioContext = React.createContext<AccordioContextValue>(contextValue);

export default AccordioContext;
