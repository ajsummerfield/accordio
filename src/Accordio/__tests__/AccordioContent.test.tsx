import React from 'react';
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { shallowRenderer } from './helpers';
import AccordioContext from '../context/AccordioContext';
import { AccordioItem, AccordioHeading, AccordioContent } from '../index';

describe('AccordioContent Component', () => {
  let component: JSX.Element;

  describe('when rendered', () => {
    beforeEach(() => {
      component = <AccordioContent>Hello World</AccordioContent>;
    });
  
    it('Should match the snapshot', () => {
      const tree = shallowRenderer(component);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when the activeItem in context matches the items index', () => {
    const context = {
      activeIndex: 0,
      setActiveIndex: jest.fn(),
    };
    
    beforeEach(() => {
      component = (
        <AccordioContext.Provider value={context}>
          <AccordioItem index={0}>
            <AccordioHeading>Heading 1</AccordioHeading>
            <AccordioContent>Content 1</AccordioContent>
          </AccordioItem>
          <AccordioItem index={1}>
            <AccordioHeading>Heading 2</AccordioHeading>
            <AccordioContent>Content 2</AccordioContent>
          </AccordioItem>
        </AccordioContext.Provider>
      );
    });

    it('should set the first item as active', () => {
      const { container } = render(component);
      expect(container.firstChild).not.toHaveStyle('max-height: 0');
    });
  });
});
