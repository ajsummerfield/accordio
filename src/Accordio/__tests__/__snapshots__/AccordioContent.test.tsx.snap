// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordioContent Component when rendered Should match the snapshot 1`] = `
<div
  className="accordio__panel"
  style={
    Object {
      "maxHeight": "0px",
    }
  }
>
  <div
    className="accordio__content"
  >
    Hello World
  </div>
</div>
`;
